### PR TITLE
Peppermint

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -478,7 +478,7 @@ func DefaultConsensusConfig() *ConsensusConfig {
 		TimeoutCommit:               1000,
 		SkipTimeoutCommit:           false,
 		CreateEmptyBlocks:           true,
-		CreateEmptyBlocksInterval:   0,
+		CreateEmptyBlocksInterval:   5000,
```
Comment (jdkanani) [inline config/config.go]: Let's increase to somewhere 120 * 60 !?

File: consensus/state.go [hunk 2]
```diff
@@ -775,6 +776,7 @@ func (cs *ConsensusState) needProofBlock(height int64) bool {
 	}
 
 	lastBlockMeta := cs.blockStore.LoadBlockMeta(height - 1)
+	cs.Logger.Info("[STATE] app hash", "apphash", hex.EncodeToString(cs.state.AppHash), "lastBlockMeta", hex.EncodeToString(lastBlockMeta.Header.AppHash))
```
Comment (jdkanani) [inline consensus/state.go]: Do we need this log?

File: consensus/state.go [hunk 3]
```diff
@@ -1268,12 +1278,22 @@ func (cs *ConsensusState) finalizeCommit(height int64) {
...
+				cs.Logger.Info(fmt.Sprintf("[peppermint] Committed vote:: Height: %v, Round: %v, VoteData %v, Sig %v", precommit.Height, precommit.Round, hex.EncodeToString(precommit.SignBytes(cs.state.ChainID)), hex.EncodeToString(precommit.Signature)))
+			}
+		}
 		cs.blockStore.SaveBlock(block, blockParts, seenCommit)
 	} else {
 		// Happens during replay if we already saved the block but didn't commit
 		cs.Logger.Info("Calling finalizeCommit on already stored block", "height", block.Height)
 	}
 
+	cs.Logger.Info(fmt.Sprintf("%v", block))
```
Comment (jdkanani) [inline consensus/state.go]: Needed?

File: consensus/state.go [hunk 4]
```diff
@@ -1644,7 +1664,14 @@ func (cs *ConsensusState) signVote(type_ byte, hash []byte, header types.PartSet
 		Type:             type_,
 		BlockID:          types.BlockID{hash, header},
 	}
+
+	if cs.LockedBlock != nil {
+		cs.Logger.Info("[peppermint] sign add vote", "lockedHeaderHash", hex.EncodeToString(header.Hash), "lockedBlockDataHash", hex.EncodeToString(cs.LockedBlock.DataHash))
```
Comment (jdkanani) [inline consensus/state.go]: Do we need this log?

File: consensus/state.go [hunk 5]
```diff
@@ -1644,7 +1664,14 @@ func (cs *ConsensusState) signVote(type_ byte, hash []byte, header types.PartSet
...
+		vote.Data = cs.LockedBlock.DataHash
+	}
+
 	err := cs.privValidator.SignVote(cs.state.ChainID, vote)
+	cs.Logger.Info("[peppermint] vote sign with data", "signBytes", vote.SignBytes(cs.state.ChainID))
```
Comment (jdkanani) [inline consensus/state.go]: Change this to debug.

File: privval/priv_validator.go [hunk 6]
```diff
@@ -158,6 +159,13 @@ func (pv *FilePV) SignVote(chainID string, vote *types.Vote) error {
 	return nil
 }
 
+
+
+
+
+
+
+
```
Comment (jdkanani) [inline privval/priv_validator.go]: Do we need spaces? :)

File: privval/priv_validator.go [hunk 7]
```diff
@@ -217,10 +225,7 @@ func (pv *FilePV) signVote(chainID string, vote *types.Vote) error {
 	if sameHRS {
 		if bytes.Equal(signBytes, pv.LastSignBytes) {
 			vote.Signature = pv.LastSignature
-		} else if timestamp, ok := checkVotesOnlyDifferByTimestamp(pv.LastSignBytes, signBytes); ok {
-			vote.Timestamp = timestamp
-			vote.Signature = pv.LastSignature
-		} else {
+		}else {
```
Comment (jdkanani) [inline privval/priv_validator.go]: Lint please.

File: rpc/core/mempool.go [hunk 8]
```diff
@@ -151,6 +152,7 @@ func BroadcastTxSync(tx types.Tx) (*ctypes.ResultBroadcastTx, error) {
 // | tx        | Tx   | nil     | true     | The transaction |
 func BroadcastTxCommit(tx types.Tx) (*ctypes.ResultBroadcastTxCommit, error) {
 	// subscribe to tx being committed in block
+	logger.Info("The transaction recieved is ",hex.EncodeToString(tx),"transaction")
```
Comment (jdkanani) [inline rpc/core/mempool.go]: Change it to Debug? and lint please.

File: state/execution.go [hunk 9]
```diff
@@ -86,6 +86,7 @@ func (blockExec *BlockExecutor) ApplyBlock(state State, blockID types.BlockID, b
 	fail.Fail() // XXX
 
 	// update the state with the block and responses
+	fmt.Errorf("abci resposnse %v",abciResponses)
```
Comment (jdkanani) [inline state/execution.go]: Please remove

File: state/execution.go [hunk 10]
```diff
@@ -276,9 +277,10 @@ func getBeginBlockValidatorInfo(block *types.Block, lastValSet *types.ValidatorS
 func updateValidators(currentSet *types.ValidatorSet, abciUpdates []abci.Validator) error {
 	updates, err := types.PB2TM.Validators(abciUpdates)
 	if err != nil {
+		fmt.Errorf("error in update validators ")
 		return err
 	}
-
+	fmt.Println("No error in getting validator updates , updates := %v",updates)
```
Comment (jdkanani) [inline state/execution.go]:
```suggestion
	fmt.Println("Validator updated successfully. Updates: %v",updates)
```

File: types/block.go [hunk 11]
```diff
@@ -210,6 +210,7 @@ func (h *Header) Hash() cmn.HexBytes {
 	if h == nil || len(h.ValidatorsHash) == 0 {
 		return nil
 	}
+	//todo maybe add votes here /maybe not
```
Comment (jdkanani) [inline types/block.go]:
```suggestion
	// TODO maybe add votes here
```

File: types/genesis.go [hunk 12]
```diff
@@ -51,18 +52,18 @@ func (genDoc *GenesisDoc) ValidatorHash() []byte {
 // ValidateAndComplete checks that all necessary fields are present
 // and fills in defaults for optional fields left empty
 func (genDoc *GenesisDoc) ValidateAndComplete() error {
-
+	fmt.Printf("here in validate and complete with gen doc as %v",genDoc.ConsensusParams)
```
Comment (jdkanani) [inline types/genesis.go]: Please remove

File: types/genesis.go [hunk 13]
```diff
@@ -51,18 +52,18 @@ func (genDoc *GenesisDoc) ValidatorHash() []byte {
...
 	}
-
-	if genDoc.ConsensusParams == nil {
-		genDoc.ConsensusParams = DefaultConsensusParams()
-	} else {
-		if err := genDoc.ConsensusParams.Validate(); err != nil {
-			return err
-		}
-	}
+	genDoc.ConsensusParams = DefaultConsensusParams()
```
Comment (jdkanani) [inline types/genesis.go]: What is this for?

File: types/genesis.go [hunk 14]
```diff
@@ -91,8 +92,9 @@ func GenesisDocFromJSON(jsonBlob []byte) (*GenesisDoc, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	if err := genDoc.ValidateAndComplete(); err != nil {
+		fmt.Println("___> here")
```
Comment (jdkanani) [inline types/genesis.go]: Please remove

File: types/params.go [hunk 15]
```diff
@@ -89,6 +90,7 @@ func DefaultEvidenceParams() EvidenceParams {
 func (params *ConsensusParams) Validate() error {
 	// ensure some values are greater than 0
 	if params.BlockSize.MaxBytes <= 0 {
+		fmt.Println(params.BlockSize)
```
Comment (jdkanani) [inline types/params.go]: Please remove

File: types/protobuf.go [hunk 16]
```diff
@@ -164,7 +169,9 @@ type pb2tm struct{}
 func (pb2tm) PubKey(pubKey abci.PubKey) (crypto.PubKey, error) {
 	// TODO: define these in crypto and use them
 	sizeEd := 32
-	sizeSecp := 33
+	//sizeSecp := 33
+	sizeSecp := 65
+	fmt.Errorf("****** the type observed is*********")
```
Comment (jdkanani) [inline types/protobuf.go]: Please remove this `Errorf`

File: types/protobuf.go [hunk 17]
```diff
@@ -164,7 +169,9 @@ type pb2tm struct{}
 func (pb2tm) PubKey(pubKey abci.PubKey) (crypto.PubKey, error) {
 	// TODO: define these in crypto and use them
 	sizeEd := 32
-	sizeSecp := 33
+	//sizeSecp := 33
+	sizeSecp := 65
```
Comment (jdkanani) [inline types/protobuf.go:172]: May be add `// [peppermint]` !?

File: types/tx.go [hunk 18]
```diff
@@ -18,6 +18,8 @@ type Tx []byte
 
 // Hash computes the TMHASH hash of the wire encoded transaction.
 func (tx Tx) Hash() []byte {
+
+	fmt.Printf("********* Inside TM Hash with tx as %v and string ", tx)
```
Comment (jdkanani) [inline types/tx.go]: please remove :)

**Review by jdkanani — CHANGES_REQUESTED** [review]

File: types/genesis.go (hunk 13 above)
Comment (vaibhavchellani) [reply types/genesis.go]: setting the default consensus params

File: config/config.go (hunk 1 above)
Comment (vaibhavchellani) [reply config/config.go]: done :)

**Comment by vaibhavchellani** [comment]
Updates have been made .

File: privval/priv_validator.go [hunk 19]
```diff
@@ -300,6 +298,16 @@ func (pv *FilePV) SignHeartbeat(chainID string, heartbeat *types.Heartbeat) erro
 	return nil
 }
 
+func (pv *FilePV) SignHello(chainID string, data string) ([]byte, error) {
```
Comment (jdkanani) [inline privval/priv_validator.go]: Do we need this?

File: types/genesis.go (hunk 13 above)
Comment (jdkanani) [reply types/genesis.go]:
```golang
if genDoc.ConsensusParams == nil {
    genDoc.ConsensusParams = DefaultConsensusParams()
} else {
	if err := genDoc.ConsensusParams.Validate(); err != nil {
	    return err
	}
}
```

no?

**Review by jdkanani — CHANGES_REQUESTED** [review]

**Review by jdkanani — APPROVED** [review]

**Pull request merged**